### PR TITLE
Fix For Maui WinUI Registrations

### PR DESCRIPTION
**Pull request by ChrisPulman**
Fix for #3576

**What kind of change does this PR introduce?**

Fix for #3576 

**What is the current behaviour?**

ActivationForViewFetcher is not registered

**What is the new behaviour?**

Maui and WinUI Registrations are now Registered
Namespace issue resolved

**What might this PR break?**

None

**Please check if the PR fulfils these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:

### CODE REVIEW

File: src/ReactiveUI.Maui/ActivationForViewFetcher.cs [hunk 1]
```diff
@@ -28,12 +33,15 @@ public class ActivationForViewFetcher : IActivationForViewFetcher
     /// <inheritdoc/>
     public int GetAffinityForView(Type view) =>
 #if HAS_WINUI
+#if IS_MAUI
```
Comment (anaisbetts) [inline src/ReactiveUI.Maui/ActivationForViewFetcher.cs:36]: Should IS_MAUI be nested in the HAS_WINUI here?

File: src/ReactiveUI.Maui/ActivationForViewFetcher.cs [hunk 2]
```diff
@@ -55,9 +66,10 @@ public IObservable<bool> GetActivationForView(IActivatableView view)
         return activation.DistinctUntilChanged();
     }
 
-    private static IObservable<bool>? GetActivationFor(ICanActivate? canActivate) => canActivate?.Activated.Select(_ => true).Merge(canActivate.Deactivated.Select(_ => false));
+    private static IObservable<bool>? GetActivationFor(ICanActivate? canActivate) =>
+        canActivate?.Activated.Select(_ => true).Merge(canActivate.Deactivated.Select(_ => false));
 
-#if HAS_MAUI
+#if HAS_MAUI || (HAS_WINUI && IS_MAUI)
```
Comment (anaisbetts) [inline src/ReactiveUI.Maui/ActivationForViewFetcher.cs:72]: What's the difference between HAS_MAUI and IS_MAUI? Should we rename these?

**Review by anaisbetts — CHANGES_REQUESTED** [review]

File: src/ReactiveUI.Maui/ActivationForViewFetcher.cs (hunk 2 above)
Comment (ChrisPulman) [reply src/ReactiveUI.Maui/ActivationForViewFetcher.cs:72]: These are to distinguish the Maui Android / iOS section (HAS_MAUI) and the Maui WinUI section HAS_WinUI, as we are using a single file between the two projects (Maui and WinUI) the namespaces need to match the assembly hence the IS_MAUI and IS_WINUI relates to the Project SDK values of UseMaui and UseWinUI.

**Review by anaisbetts — APPROVED** [review]

File: src/ReactiveUI.Maui/ActivationForViewFetcher.cs (hunk 2 above)
Comment (glennawatson) [reply src/ReactiveUI.Maui/ActivationForViewFetcher.cs:72]: Is there a way for us to have more meaningful name for one of them. Don't mind a bit of verbosity.

Comment (ChrisPulman) [reply src/ReactiveUI.Maui/ActivationForViewFetcher.cs:72]: I can look at dropping HAS_MAUI and invert HAS_WINUI where required, perhaps make HAS_WINUI => WINUI_TARGET

Comment (glennawatson) [reply src/ReactiveUI.Maui/ActivationForViewFetcher.cs:72]: Yeah seems nicer name.

**Pull request merged**